### DISCUSSION TITLE
vs code integrated terminal opens wrong container

### DISCUSSION THREAD

**Discussion by hannes09** [coder/coder#8666] [General] [answered]
Hi,

i have two different templates based on the [docker example](https://github.com/coder/coder/blob/main/examples/templates/docker/main.tf).

For each template I have one workspace.

When i run both workspaces at the same time the integrated terminal from vs code always opens/connects to the container from the workspace that was started first. So I end up in the wrong container for the second workspace. 

Each  template/workspace works fine if no other workspace is running, but I can not run both at the same time.
The terminal from coder (outside vs code) always opens in the correct container, only the integrated terminal form vs code is affected.

**Comment by matifali** [comment]
That's pretty strange. Can you share your exact template?

**Reply by hannes09** [reply, marked as answer]
I found a workaround for this.
If I use different urls in the workspace template the terminal works.

`url          = "http://localhost:13337/?folder=/home/${local.username}"`

and for the other template

`url          = "http://localhost:13338/?folder=/home/${local.username}"`
